src/Authorization.AspNetCore/AuthorizationValidationRule.cs

```diff
@@ -68,108 +71,120 @@ public Task<INodeVisitor> ValidateAsync(ValidationContext context)
+            else if (policyNames?.Count > 1)
             {
-                var task = _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, policyName);
-                tasks.Add(task);
+                var tasks = new List<Task<AuthorizationResult>>(policyNames.Count);
+                foreach (string policyName in policyNames)
+                {
+                    var task = _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, policyName);
+                    tasks.Add(task);
+                }
+
+                var authorizationResults = await Task.WhenAll(tasks);
+
+                foreach (var result in authorizationResults)
+                {
+                    AddValidationError(node, context, operationType, result);
+                }
             }
-            await Task.WhenAll(tasks);
 
-            foreach (var task in tasks)
+            static void AddValidationError(INode node, ValidationContext context, OperationType? operationType, AuthorizationResult result)
             {
-                var result = task.Result;
                 if (!result.Succeeded)
                 {
-                    var stringBuilder = new StringBuilder("You are not authorized to run this ");
-                    stringBuilder.Append(operationType.ToString().ToLower());
-                    stringBuilder.AppendLine(".");
+                    var error = new StringBuilder("You are not authorized to run this ")
+                                .Append(GetOperationType(operationType))
+                                .Append(".");
 
                     foreach (var failure in result.Failure.FailedRequirements)
                     {
-                        AppendFailureLine(stringBuilder, failure);
+                        AppendFailureLine(error, failure);
                     }
 
-                    context.ReportError(new ValidationError(context.OriginalQuery, "6.1.1", stringBuilder.ToString(), node)
+                    context.ReportError(new ValidationError(context.OriginalQuery, "6.1.1", error.ToString(), node == null ? Array.Empty<INode>() : new INode[] { node })
                     {
                         Code = "authorization",
                     });
                 }
             }
         }
 
-        private static void AppendFailureLine(
-            StringBuilder stringBuilder,
-            IAuthorizationRequirement authorizationRequirement)
+        private static string GetOperationType(OperationType? operationType)
         {
+            return operationType switch
+            {
+                OperationType.Query => "query",
+                OperationType.Mutation => "mutation",
+                OperationType.Subscription => "subscription",
+                _ => "operation",
+            };
+        }
+
+        private static void AppendFailureLine(StringBuilder error, IAuthorizationRequirement authorizationRequirement)
+        {
+            error.AppendLine();
```

Comment:
I have changed the sequence of `AppendLine` calls for carriage return.

Before:
```
You are not authorized to run this operation.
Required claim 'some' is not present.
                                                   <-------------- NEW LINE HERE
```

After:
```
You are not authorized to run this operation.
Required claim 'some' is not present.              <-------------- NO NEW LINE
```